ts/private/ts_lib.bzl

```diff
@@ -87,6 +87,18 @@ https://docs.aspect.build/rulesets/aspect_rules_js/docs/js_library#deps for more
         mandatory = True,
         allow_single_file = [".json"],
     ),
+    "async_typecheck": attr.bool(
```

Comment:
I'm not in love with this name, though we will stick it in a macro anyway so it's not a huge deal :)

For us we did a single flag `enforce_isolated_declarations` that enabled this mode + isoDecls in the tsconfig. Do you think there's any case where someone would set isoDecls in tsconfigs but not want to enable this?

Reply:
Lets move this discussion here: https://github.com/aspect-build/rules_ts/pull/679#discussion_r1759506820